core/invoker/src/main/scala/org/apache/openwhisk/core/containerpool/ContainerProxy.scala

```diff
@@ -770,6 +770,23 @@ object ContainerProxy {
           causedBy ++ initTime ++ binding
       })
   }
+
+  /**
+   * Partitions the activation arguments into two JsObject instances. The first is exported as intended for export
+   * by the action runtime to the environment. The second is passed on as arguments to the action.
+   *
+   * @param content the activation arguments
+   * @param initArgs set of parameters to treat as initialization arguments
+   * @return A partition of the arguments into an environment variables map and the JsObject argument to the action
+   */
+  def partitionArguments(content: Option[JsObject], initArgs: Set[String]): (Map[String, JsValue], JsObject) = {
```

Comment:
May be to avoid `content.get` i.e. get on option (though its safe here) we can use

```scala
  def partitionArguments(content: Option[JsObject], initArgs: Set[String]): (Map[String, JsValue], JsObject) = {
    content match {
      case Some(js) if initArgs.empty => (Map.empty, js)
      case Some(js) =>
        val (env, args) = js.fields.partition(k => initArgs.contains(k._1))
        (env, JsObject(args))
      case None =>
        (Map.empty, JsObject.empty)
    }
```

Reply:
Further it would be better if we restrict the env to `Map[String, String]` here i.e. restict/coerce the value to string here

Reply:
I'll defer this to the `containerInitializer()`.

Reply:
OK - Since we have to turn this into a JsObject, we eventually still need a Map[String, JsString].